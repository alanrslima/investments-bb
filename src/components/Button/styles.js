import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 20,
    backgroundColor: colors.YELLOW,
  },
  text: {
    fontWeight: 'bold',
    color: colors.BLUE,
    fontSize: fonts.DESCRIPTION
  }
});
