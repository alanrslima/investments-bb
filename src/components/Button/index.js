import React from 'react';
import { TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

function Button({ text, onPress, containerStyle }) {
  return (
    <TouchableOpacity style={[styles.container, containerStyle]} onPress={onPress}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  containerStyle: PropTypes.objectOf(PropTypes.number)
};

Button.defaultProps = {
  containerStyle: {}
};

export default Button;
