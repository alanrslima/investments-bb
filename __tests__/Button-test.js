import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../src/components/Button';


test('renders correctly', () => {
  const tree = renderer.create(<Button
    text="Text button"
    onPress={() => { }}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
