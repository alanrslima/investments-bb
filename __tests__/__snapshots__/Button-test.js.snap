// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F6E155",
      "justifyContent": "center",
      "opacity": 1,
      "paddingVertical": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#225AA0",
        "fontSize": 15,
        "fontWeight": "bold",
      }
    }
  >
    Text button
  </Text>
</View>
`;
